### ISSUE TITLE
more a personal problem

### ISSUE THREAD

**Issue by RhythmicGaming** [MagicTheGathering/mtg-sdk-python#32]
does anyone have ideas on parsing card text for synergies ive used the sdk to create a database in hopes i can find a player bot ai to train with all these cards but basically im looking for anything anyone has done with this apk in the way of development, the project i hope to one day complete is fairly feature heavy and i could use all the help i can get, tysm for this library its been incredibly helpful

**Comment by hitz** [comment]
you could try manually curating synergy between cards (just rate each entry in the matrix on like a 0-5 scale) - do this for one set or a partial set and try to train a ML alg to predict it for a different set.
Might also help you come up with some heuristics?

**Comment by RhythmicGaming** [comment]
The issue with manual curation is at that point I'd basically be building
every decklist possible, manually, which is something like 400billion
combinations possible, I'm looking into training an ai to recognize
interactions and text implications at the moment, if I could get the bot to
identify synergies and label them for me, I think what youre saying could
work.

**Comment by hitz** [comment]
no you just approximate it with a pairwise matrix.   Like if a card says "other goblins you control get +1/+1" then you give a 4 to every other card with creature type goblin. 
It's something like 250x250 interactions but there are a lot of zeros.

But you can't train anything without a training set.

**Comment by RhythmicGaming** [comment]
Very true, I figured something similar could be used for tribal
interactions, its everything else I feel overwealmed by, specifically etb
effects and non creature interactions, these exceptions are a little
complex for me to try to make any generalizations about, not all of them
are even possitive things in some cases, which is also dependant on synergy
and at times even game state.

**Comment by miguelBra** [comment]
The best way I think its to develop a graph neural network for authomatic embedding of card features. For a very detailed ability learning you could use a recurrent neural network but that could be too difficult, considering you would also need a nice embedding for starting processing words. 
I just discovered this api and I think I will start also trying to fit a graph (in case i am able to understand how to manipulate the querysets... no success so far).
